### PR TITLE
Optimized at_c and back.

### PR DESCRIPTION
**Pull request by jonathanpoelen**
Small optimization to `at_c`.
Removal `detail :: last element` (only used by` detail :: back_impl`). As stated in #128, a specialized version is faster than `at_c`.

```cpp
#include <brigand/sequences/list.hpp>
#include <brigand/sequences/range.hpp>
#include <brigand/sequences/at.hpp>

template<int N, int I>
constexpr bool test_at_c()
{
  using list = brigand::range<long, 0, N>;
  return brigand::at_c<list, I>{},1;
}

static_assert(test_at_c<1,0>(), "");
static_assert(test_at_c<10,0>(), "");
static_assert(test_at_c<10,1>(), "");
static_assert(test_at_c<10,2>(), "");
static_assert(test_at_c<10,3>(), "");
static_assert(test_at_c<10,4>(), "");
static_assert(test_at_c<10,5>(), "");
static_assert(test_at_c<10,6>(), "");
static_assert(test_at_c<10,7>(), "");
static_assert(test_at_c<10,8>(), "");
static_assert(test_at_c<10,9>(), "");
static_assert(test_at_c<100,0>(), "");
static_assert(test_at_c<100,50>(), "");
static_assert(test_at_c<100,99>(), "");
static_assert(test_at_c<500,0>(), "");
static_assert(test_at_c<500,100>(), "");
static_assert(test_at_c<500,200>(), "");
static_assert(test_at_c<500,300>(), "");
static_assert(test_at_c<500,400>(), "");
static_assert(test_at_c<500,499>(), "");
// old: 0:00.49s  98628k
// new: 0:00.37s  75452k

static_assert(test_at_c<1000,0>(), "");
static_assert(test_at_c<2000,0>(), "");
static_assert(test_at_c<3000,0>(), "");
//old: 0:03.93s  634116k
//new: 0:03.46s  591740k

static_assert(test_at_c<2000,1000>(), "");
static_assert(test_at_c<3000,1500>(), "");
//old: 0:05.49s  855400k
//new: 0:04.29s  772692k

static_assert(test_at_c<1000,500>(), "");
static_assert(test_at_c<1000,999>(), "");
static_assert(test_at_c<2000,1500>(), "");
static_assert(test_at_c<2000,1999>(), "");
static_assert(test_at_c<3000,2500>(), "");
static_assert(test_at_c<3000,2999>(), "");
//old: 0:11.45s  1703284k
//new: 0:09.35s  1570256k
```

### CODE REVIEW

File: brigand/sequences/at.hpp [hunk 1]
```diff
@@ -9,28 +9,26 @@
...
-    template<template<typename...> class L, typename... N>
+    template<template<typename...> class L, class... N>
     struct element_at<L<N...>>
     {
-      template<typename T> type_<T> operator()(ignore<N>..., type_<T>, ...);
+      template<typename T> T static at(typename voidp<N>::type..., T*, ...);
     };
 
-    template<std::size_t N, typename Seq> struct at_impl;
+    template<int N, typename Seq> struct at_impl;
```
Comment (jfalcou) [inline brigand/sequences/at.hpp]: why the change to int ?

File: brigand/types/voidp.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  Copyright (c) 2015 Edouard Alligand and Joel Falcou
+
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+=================================================================================================**/
+#pragma once
+
+namespace brigand
+{
+  template<typename T> struct voidp { using type = void const *; };
```
Comment (jfalcou) [inline brigand/types/voidp.hpp]: can't it be an alias ?

File: brigand/sequences/at.hpp (hunk 1 above)
Comment (jonathanpoelen) [inline brigand/sequences/at.hpp]: I do not know, probably a bad reason: /.

File: brigand/types/voidp.hpp (hunk 2 above)
Comment (jonathanpoelen) [inline brigand/types/voidp.hpp]: problem with msvs (see #128)

**Comment by porkybrain** [comment]
might be another #ifdef candidate. MSVC2013's alias support sucks!
